Controllers/SightingControllers.js

```diff
@@ -0,0 +1,30 @@
+class SightingsController {
+  constructor(getSightings) {
+    this.getSightings = getSightings;
+  }
+
+  getAllData = async (req, res) => {
+    const sightings = await this.getSightings();
+    res.json(sightings);
+  };
+
+  getOneData = async (req, res) => {
+    const { sightingIndex } = req.params;
+    if (isNaN(Number(sightingIndex))) {
+      return res.status(400).send({ error: true, msg: "Wrong Type" });
+    }
+    const sightings = await this.getSightings();
+    res.json(sightings[sightingIndex]);
+  };
+
+  getFilterData = async (req, res) => {
+    const { filter, filterData } = req.params;
+    const sightings = await this.getSightings();
+    const output = sightings.filter(
+      (data) => data[filter] && data[filter].toLowerCase().includes(filterData)
+    );
```

Comment:
You could make this operation more efficient by running the includes outside of the filter function. Here you have O(n2), whereas you could improve your algo to O(n) if you separate the two processes.

Reply:
Sorry, I misread the code here :) Guess no other choice here since we work with a string. Can ignore my previous comment